core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -2175,4 +2150,25 @@ private void validatePropertiesToSet(Map<String,String> opts, Map<String,String>
       opts.put(k, v);
     });
   }
+
+  @Override
+  public void setTabletHostingGoal(String tableName, Range range, String goal)
+      throws AccumuloSecurityException, AccumuloException, TableNotFoundException {
+    EXISTING_TABLE_NAME.validate(tableName);
+    NOT_BUILTIN_TABLE.validate(tableName);
+    checkArgument(range != null, "range is null");
+    checkArgument(goal != null, "goal is null");
+
+    TableId tableId = context.getTableId(tableName);
+    TabletHostingGoal g = TabletHostingGoal.valueOf(goal.toUpperCase());
+
+    List<TKeyExtent> extents =
+        TabletLocatorImpl.findExtentsForRange(context, tableId, range, Set.of());
+
+    log.debug("Setting tablet hosting goal to {} for extents: {}", goal, extents);
+    ThriftClientTypes.TABLET_MGMT.executeVoid(context,
+        client -> client.setTabletHostingGoal(TraceUtil.traceInfo(), context.rpcCreds(),
+            tableId.canonical(), extents, g.toThrift()));
```

Comment:
Instead of resolving the range into extents on the client side, it would probably be better to pass the range to the manager and let it do the resolution.  Could also make this a FATE operations so that the entire range is processed even in the case of failures.

Making it a FATE operation and passing the range to the manager could be follow on work/PR.